### PR TITLE
[framework] customer structure refactoring

### PR DESCRIPTION
**Pull request by TomasLudvik**
| Q             | A
| ------------- | ---
|Description, reason for the PR| There were some irregularities in managing customers and billing addresses. This has been united, so these entities are now managed via their facades.
|New feature| No 
|[BC breaks](https://docs.shopsys.com/en/latest/contributing/backward-compatibility-promise/)| Yes 
|Fixes issues| ... 
|Have you read and signed our [License Agreement for contributions](https://www.shopsys.com/license-agreement)?| Yes

### CODE REVIEW

File: packages/framework/tests/Unit/Model/Customer/CustomerUserUpdateDataFactoryTest.php [hunk 1]
```diff
@@ -38,7 +39,9 @@ public function testGetAmendedCustomerUserUpdateDataByOrderWithoutChanges()
     {
         $customerUserUpdateUpdateDataFactory = $this->getCustomerUserUpdateDataFactory();
 
-        $customer = new Customer();
+        $customerData = new CustomerData();
```
Comment (s3tezsky) [inline packages/framework/tests/Unit/Model/Customer/CustomerUserUpdateDataFactoryTest.php]: We should use `CustomerDataFactory` for creating `$customerData`. What do you say?

File: packages/framework/tests/Unit/Model/Customer/UserTest.php [hunk 2]
```diff
@@ -9,29 +9,39 @@
...
+use Shopsys\FrameworkBundle\Model\Customer\CustomerData;
 use Shopsys\FrameworkBundle\Model\Customer\User\CustomerUser;
 use Shopsys\FrameworkBundle\Model\Customer\User\CustomerUserData;
 
 class UserTest extends TestCase
 {
     public function testGetFullNameReturnsLastnameAndFirstnameForUser()
     {
-        $customer = new Customer();
+        $customerData = new CustomerData();
```
Comment (s3tezsky) [inline packages/framework/tests/Unit/Model/Customer/UserTest.php]: Use `CustomerDataFactory`

File: project-base/tests/App/Functional/PersonalData/PersonalDataExportXmlTest.php [hunk 3]
```diff
@@ -35,10 +36,17 @@ class PersonalDataExportXmlTest extends TransactionFunctionalTestCase
     public function testExportXml()
     {
         $country = $this->createCountry();
-        $customer = new Customer();
-        $customer->addBillingAddress($this->createBillingAddress($country, $customer));
+
+        $customerData = new CustomerData();
```
Comment (s3tezsky) [inline project-base/tests/App/Functional/PersonalData/PersonalDataExportXmlTest.php]: Again `CustomerDataFactory`

File: packages/framework/src/Model/Customer/Customer.php [hunk 4]
```diff
@@ -36,16 +36,50 @@ class Customer
...
+     * @ORM\Column(type="integer")
+     */
+    protected $domainId;
+
+    /**
+     * @param \Shopsys\FrameworkBundle\Model\Customer\CustomerData $customerData
+     */
+    public function __construct(CustomerData $customerData)
+    {
+        $this->setData($customerData);
```
[the remark this answers is not in the document]
Comment (TomasLudvik) [reply packages/framework/src/Model/Customer/Customer.php]: I do not like using static, plus our coding standards say that you have to create entity via its factory.

File: packages/framework/src/Model/Customer/CustomerDataFactoryInterface.php [hunk 5]
```diff
@@ -0,0 +1,23 @@
+<?php
+
```
[the remark this answers is not in the document]
Comment (TomasLudvik) [reply packages/framework/src/Model/Customer/CustomerDataFactoryInterface.php]: fixed

File: packages/framework/tests/Unit/Model/Customer/CustomerUserUpdateDataFactoryTest.php (hunk 1 above)
Comment (TomasLudvik) [reply packages/framework/tests/Unit/Model/Customer/CustomerUserUpdateDataFactoryTest.php]: I agree with that thought, but as it is Unit test you cannot get instance of factory there

File: packages/framework/tests/Unit/Model/Customer/UserTest.php (hunk 2 above)
Comment (TomasLudvik) [reply packages/framework/tests/Unit/Model/Customer/UserTest.php]: same as previous answer

File: project-base/tests/App/Functional/PersonalData/PersonalDataExportXmlTest.php (hunk 3 above)
Comment (TomasLudvik) [reply project-base/tests/App/Functional/PersonalData/PersonalDataExportXmlTest.php]: same as two previous :)

**Pull request merged**